rust/signed_doc/src/metadata/doc_type.rs

```diff
@@ -0,0 +1,225 @@
+//! Document Type.
+
+use std::fmt::{Display, Formatter};
+
+use catalyst_types::{
+    problem_report::ProblemReport,
+    uuid::{CborContext, Uuid, UuidV4},
+};
+use minicbor::{Decode, Decoder, Encode};
+use tracing::warn;
+
+use crate::decode_context::{ConversionPolicy, DecodeContext};
+
+/// List of `UUIDv4` document type.
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct DocType(Vec<UuidV4>);
+
+impl DocType {
+    /// Get a list of `UUIDv4` document types.
+    #[allow(dead_code)]
+    pub fn doc_types(&self) -> &Vec<UuidV4> {
+        &self.0
+    }
+}
+
+impl From<UuidV4> for DocType {
+    fn from(value: UuidV4) -> Self {
+        DocType(vec![value])
+    }
+}
+
+impl TryFrom<Uuid> for DocType {
+    type Error = anyhow::Error;
+
+    fn try_from(value: Uuid) -> Result<Self, Self::Error> {
+        let uuid_v4 = UuidV4::try_from(value)?;
+        Ok(DocType(vec![uuid_v4]))
+    }
+}
+
+impl From<Vec<UuidV4>> for DocType {
+    fn from(value: Vec<UuidV4>) -> Self {
```

Comment:
This should be try from, and it should fail if the vec is empty.  Its invalid to create a DocType with no values.
Then the encoder can't fail because its not possible to create the type such that it can't be encoded.